crates/replace/src/lib.rs

```diff
@@ -275,23 +275,29 @@ pub mod pallet {
         ///   - `B` is `transaction` size in bytes (length-fee-bounded)
         #[pallet::call_index(3)]
         #[pallet::weight({
-            let h = merkle_proof.hashes.len() as u32;
-            let i = transaction.inputs.len() as u32;
-            let o = transaction.outputs.len() as u32;
-            let b = *length_bound;
+            let h = transaction.user_tx_proof.merkle_proof.hashes.len() as u32;
+            let i = transaction.user_tx_proof.transaction.inputs.len() as u32;
+            let o = transaction.user_tx_proof.transaction.outputs.len() as u32;
+            let b = transaction.user_tx_proof.tx_encoded_len;
             <T as Config>::WeightInfo::execute_pending_replace(h, i, o, b)
                 .max(<T as Config>::WeightInfo::execute_cancelled_replace(h, i, o, b))
-        })]
+        }.saturating_add({
+            let h = transaction.coinbase_proof.merkle_proof.hashes.len() as u32;
+            let i = transaction.coinbase_proof.transaction.inputs.len() as u32;
+            let o = transaction.coinbase_proof.transaction.outputs.len() as u32;
+            let b = transaction.coinbase_proof.tx_encoded_len;
+            <T as Config>::WeightInfo::execute_pending_replace(h, i, o, b)
+                .max(<T as Config>::WeightInfo::execute_cancelled_replace(h, i, o, b))
```

Comment:
Might it be cleaner to extract this into a function? Same for `issue` and `redeem`